db/queries/operations/process.py

```diff
@@ -0,0 +1,24 @@
+from db.transforms.base import PossiblyPartialTransform
+
+
+def get_processed_transformations(db_query):
+    """
+    Processes db_query's transformations and returns them. Resulting sequence is the db_query's
+    transform sequence, but each possibly partial transform is replaced with a transform that's the
+    result of processing it.
+
+    See PossiblyPartialTransform for more information.
+    """
+    return tuple(
+        _get_processed_transformation(db_query, ix, db_transformation)
+        for ix, db_transformation
+        in enumerate(db_query.transformations)
+    )
+
+
+def _get_processed_transformation(db_query, ix, db_transformation):
+    if isinstance(db_transformation, PossiblyPartialTransform):
+        db_transformation = db_transformation.get_processed(
+            db_query, ix_in_transform_pipeline=ix
+        )
+    return db_transformation
```

Comment:
This method bakes in a requirement for every transform that we'd want to process, along with an assumption about the needed input for processing. 
- We may want to process the same transform in multiple ways. 
- We may need different input for different transform types' processing, or for different ways to process the same transform. Etc.


This should just call something like the following method of the `DBQuery` class:
```python
    def finish_summarize_transforms(self):
        for transform, ix in enumerate(db_query.transformations):
            if isinstance(transform, Summarize):
                transform = finish_specifying_summarize_transform(self, ix, self.engine, self.metadata)
                break
        return transform
```

where the `finish_specifying_summarize_transform` is imported from the code in your other PR.

Reply:
Your proposal makes sense, especially since we don't have other partial transforms planned. Note that it's a little more work to refactor than just remove the `PossiblyPartialTransform` class: relevant tests are currently based around it.